tools/topology/platform/intel/all.m4

```diff
@@ -0,0 +1,28 @@
+ifelse(PLATFORM, `cht', include(`platform/intel/cht.m4'),
+    ifelse(PLATFORM, `byt', include(`platform/intel/byt.m4'),
+    ifelse(PLATFORM, `bdw', include(`platform/intel/bdw.m4'),
+    ifelse(PLATFORM, `cht', include(`platform/intel/cht.m4'),
+    ifelse(PLATFORM, `bxt', include(`platform/intel/bxt.m4'),
+    ifelse(PLATFORM, `hsw', include(`platform/intel/hsw.m4'),
+    ifelse(PLATFORM, `whl', include(`platform/intel/cnl.m4'),
+    ifelse(PLATFORM, `cml', include(`platform/intel/cnl.m4'),
+    ifelse(PLATFORM, `icl', include(`platform/intel/icl.m4'),
+    ifelse(PLATFORM, `tgl', include(`platform/intel/icl.m4'),
+    ifelse(PLATFORM, `ehl', include(`platform/intel/icl.m4'),
+    `')))))))))))
+
+# config SSP INDEX NAME MCLK
+undefine(`SSP_INDEX')
+define(`SSP_INDEX', ifelse(PLATFORM, `icl', `0',
+        ifelse(PLATFORM, `whl', `1',
+                ifelse(PLATFORM, `cml', `0', `'))))
+
+undefine(`SSP_NAME')
+define(`SSP_NAME', ifelse(PLATFORM, `icl', `SSP0-Codec',
+        ifelse(PLATFORM, `whl', `SSP1-Codec',
+                ifelse(PLATFORM, `cml', `SSP0-Codec', `'))))
+
+undefine(`SSP_MCLK_RATE')
+define(`SSP_MCLK_RATE', ifelse(PLATFORM, `icl', `19200000',
+        ifelse(PLATFORM, `whl', `24000000',
+                ifelse(PLATFORM, `cml', `24000000', `'))))
```

Comment:
same here, this is either too limited or too broad. And it's not really good since on WHK/CML the mclk is fixed but on APL/ICL/TGL  it's programmable